src/geff/_graph_libs/_backend_protocol.py

```diff
@@ -0,0 +1,150 @@
+
+        Args:
+            metadata (GeffMetadata): The metadata of the graph.
+            node_ids (NDArray[Any]): An array containing the node ids. Must have same dtype as
+                edge_ids.
+            edge_ids (NDArray[Any]): An array containing the edge ids. Must have same dtype
+                as node_ids.
+            node_props (dict[str, PropDictNpArray]): A dictionary
+                from node property names to (values, missing) arrays, which should have same
+                length as node_ids. Spatial graph does not support missing attributes, so the
+                missing arrays should be None or all False. If present, the missing arrays are
+                ignored with warning
+            edge_props (dict[str, PropDictNpArray]): A dictionary
+                from edge property names to (values, missing) arrays, which should have same
+                length as edge_ids. Spatial graph does not support missing attributes, so the
+                missing arrays should be None or all False. If present, the missing array is ignored
+                with warning.
+            *args (Any): Additional positional arguments used for construction.
+            **kwargs (Any): Additional keyword arguments used for construction.
+
+        Returns:
+            graph (T): The graph object.
+        """
+        ...
+
+    @classmethod
+    def read(
+        cls,
+        store: StoreLike,
+        structure_validation: bool = True,
+        node_props: list[str] | None = None,
+        edge_props: list[str] | None = None,
+        data_validation: ValidationConfig | None = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> tuple[T, GeffMetadata]:
+        """
+        Read a GEFF to a chosen backend graph instance.
+
+        Args:
+            store (StoreLike): The path or zarr store to the root of the geff zarr, where
+                the .attrs contains the geff  metadata.
+            structure_validation (bool, optional): Flag indicating whether to perform validation
+                on the geff file before loading into memory. If set to False and there are format
+                issues, will likely fail with a cryptic error. Defaults to True.
+            node_props (list of str, optional): The names of the node properties to load,
+                if None all properties will be loaded, defaults to None.
+            edge_props (list of str, optional): The names of the edge properties to load,
+                if None all properties will be loaded, defaults to None.
+            data_validation (ValidationConfig, optional): Optional configuration for which
+                optional types of data to validate. Each option defaults to False.
+            *args (Any): Additional args that may be accepted by the backend when reading the
+                data.
+            **kwargs (Any): Additional kwargs that may be accepted by the backend when reading the
+                data.
```

Comment:
somewhat unfortunate to have to duplicate all of these docstrings between both the backend classes, and the public api functions.  If we commit to making the backend objects just internal implementation details, you could consider deleting these arguments (so they don't have to be manually kept in sync with the function docstrings) and simply leave a note to backend-creating developers to see the docstrings for the corresponding functions in `_api_wrapper`?

(but, just keeping them in sync is also fine)